src/main/java/io/jenkins/pluginhealth/scoring/probes/SCMLinkValidationProbe.java

```diff
@@ -0,0 +1,121 @@
+
+/**
+ * Analyzes the {@link Plugin#getScm()} value to determine if the link is valid or not.
+ */
+@Component
+@Order(value = SCMLinkValidationProbe.ORDER)
+public final class SCMLinkValidationProbe extends Probe {
+    private static final Logger LOGGER = LoggerFactory.getLogger(SCMLinkValidationProbe.class);
+    private static final String GH_REGEXP = "https://(?<server>[^/]*)/(?<org>[^/]*)/(?<repo>[^/]*)(?:/(?<folder>.*))??";
+    private static final Pattern GH_PATTERN = Pattern.compile(GH_REGEXP);
+
+    public static final int ORDER = 1;
+
+    private final GithubConfiguration githubConfiguration;
+
+    public SCMLinkValidationProbe(GithubConfiguration githubConfiguration) {
+        this.githubConfiguration = githubConfiguration;
+    }
+
+    @Override
+    public ProbeResult doApply(Plugin plugin) {
+        if (plugin.getScm() == null || plugin.getScm().isBlank()) {
+            LOGGER.warn("{} has no SCM link", plugin.getName());
+            return ProbeResult.failure(key(), "No SCM link");
+        }
+        return fromSCMLink(plugin.getScm())
+            .map(repo -> ProbeResult.success(key(), "SCM link is valid"))
+            .orElseGet(() -> ProbeResult.failure(key(), "SCM link is invalid"));
+    }
+
+    @Override
+    public String key() {
+        return "scm";
+    }
+
+    @Override
+    protected boolean requiresRelease() {
+        return true;
+    }
+
+    private Optional<GHRepository> fromSCMLink(@NotNull String scm) {
+        Matcher matcher = GH_PATTERN.matcher(scm);
+        if (!matcher.find()) {
+            if (LOGGER.isDebugEnabled()) {
+                LOGGER.debug("{} is not respecting the SCM URL Template", scm);
+            }
+            return Optional.empty();
+        }
+        try {
+            final GitHub gitHub = this.getGitHub();
+            if (LOGGER.isTraceEnabled()) {
+                GHRateLimit rateLimit = gitHub.getRateLimit();
+                LOGGER.trace("GitHub rate: {}/{}, reset: {}", rateLimit.getRemaining(), rateLimit.getLimit(), rateLimit.getResetDate());
+            }
+            final GHOrganization org = gitHub.getOrganization(matcher.group("org"));
+            if (!"jenkinsci".equals(org.getLogin())) {
+                LOGGER.warn("{} not in {} organization", scm, "jenkinsci");
+                return Optional.empty();
+            }
+            final GHRepository repo = org.getRepository(matcher.group("repo"));
+            if (repo == null) {
+                return Optional.empty();
+            }
+            if (matcher.group("folder") != null) {
+                LOGGER.debug("{}", matcher.group("folder"));
+                // TODO
+            }
+            return Optional.of(repo);
+        } catch (IOException ex) {
+            return Optional.empty();
+        }
+    }
+
+    private GitHub getGitHub() throws IOException {
+        return GitHub.connectUsingOAuth(githubConfiguration.getGitAccessToken());
```

Comment:
Any requests to the GitHub API requires the token. If not, then the requests are unauthenticated and you are only eligible to make 60 requests per hour, when authenticated requests are limited to 5000 per hour.

Reply:
So even with 5000 reqs per hour, we are seeing a rate limit issue?

Reply:
yes, because some requests are hidden. First I was doing "GitHub, given me this organization" and then "from this organization, give me the repository". This is at least 2 calls to GitHub. As we have 1883 plugins, it's 3766 requests. So yes, we (I) have to be careful about this.

Reply:
it's even worse than that: if you do a `curl -I https://api.github.com/repos/jenkinsci/mailer-plugin` (as in https://docs.github.com/en/rest/overview/resources-in-the-rest-api#detailed-representations), you get

```
[...]
x-ratelimit-limit: 60
x-ratelimit-remaining: 58
x-ratelimit-reset: ******
x-ratelimit-resource: core
x-ratelimit-used: 2
[...]
```

so it consumed 2 requests even though you only made one. Note the `60` is because I wasn't authenticated doing this request.